crates/api/src/values.rs

```diff
@@ -133,6 +133,13 @@ impl Val {
     pub fn unwrap_anyref(&self) -> AnyRef {
         self.anyref().expect("expected anyref")
     }
+
+    pub(crate) fn comes_from_same_store(&self, store: &Store) -> bool {
+        match self {
+            Val::FuncRef(f) => Store::same(store, f.store()),
+            _ => true,
```

Comment:
Do we need to handle `AnyRef` here too, for the `InternalRef` case?

Reply:
Currently it technically doesn't need to because we don't really support `anyref` in wasmtime at the moment. I went ahead and explicitly matched it to always return `false` with a TODO comment about how to fix it in the future.